File: apps/backend/src/EnterpriseAgentOs.Domain/features/Mcp/Interfaces/IMcpServerService.cs
namespace EnterpriseAgentOs.Domain.Features.Agents.Integrations;

public interface IIntegrationDefinitionService
{
    Task<IReadOnlyList<IntegrationDefinitionRecord>> ListAsync(Guid ownerId, CancellationToken ct = default);
    Task<IntegrationDefinitionRecord?> GetAsync(Guid ownerId, string name, CancellationToken ct = default);
    Task<IntegrationDefinitionRecord> RegisterAsync(Guid ownerId, IntegrationDefinitionRecord server, CancellationToken ct = default);
    Task DeleteAsync(Guid ownerId, string name, CancellationToken ct = default);
    Task<IReadOnlyList<IntegrationDefinitionRecord>> ListForAgentAsync(Guid agentId, Guid? ownerId = null, CancellationToken ct = default);
    Task AssignToAgentAsync(Guid agentId, string integrationName, Guid? ownerId = null, CancellationToken ct = default);
    Task UnassignFromAgentAsync(Guid agentId, string integrationName, CancellationToken ct = default);
    Task SaveCredentialAsync(Guid ownerId, string integrationName, Dictionary<string, string> fields, CancellationToken ct = default);
    Task<Dictionary<string, string>> GetDecryptedCredentialAsync(string integrationName, Guid? ownerId = null, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/mcp/IMcpServerService.java
package com.enterpriseagentos.backendjava.domain.features.mcp;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public interface IMcpServerService {
    CompletableFuture<List<McpServerRecord>> listAsync();
    CompletableFuture<McpServerRecord> getAsync(String name);
    CompletableFuture<McpServerRecord> registerAsync(McpServerRecord server);
    CompletableFuture<Void> deleteAsync(String name);
    CompletableFuture<List<McpServerRecord>> listForAgentAsync(UUID agentId);
    CompletableFuture<Void> assignToAgentAsync(UUID agentId, String serverName);
    CompletableFuture<Void> unassignFromAgentAsync(UUID agentId, String serverName);
    CompletableFuture<Void> saveCredentialAsync(String serverName, Map<String, String> fields);
    CompletableFuture<Map<String, String>> getDecryptedCredentialAsync(String serverName);
}
